### PR TITLE
Make the ErrorReporter take a SourceRange instead

### LINKED ISSUE #1938: Have ErrorReporter take a SourceRange instead of a SourcePosition
When we report errors it would be better to report the range instead of the position:

```js
function f() {
  'use strict';
  yield = 42;
}
```

Would be nice if we could report `{line: 4, column: 3}, {line: 4, column: 8}` instead.

### PR DESCRIPTION
**Pull request by arv**
This allows better error reporting in editors for example.

Fixes #1938

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/syntax/Parser.js [hunk 1]
```diff
@@ -1868,16 +1873,18 @@ export class Parser {
           result.push(new CaseClause(this.getTreeLocation_(start), expression, statements));
           break;
         }
-        case DEFAULT:
+        case DEFAULT: {
+          let defaultToken = nextToken();
           if (foundDefaultClause) {
-            this.reportError_('Switch statements may have at most one default clause');
```
Comment (johnjbarton) [inline src/syntax/Parser.js]: Maybe `\'default\'` clause

File: src/syntax/Parser.js [hunk 2]
```diff
@@ -4265,7 +4275,8 @@ export class Parser {
       declaration = this.parseStatementListItem_(type);
     }
     if (this.annotations_.length > 0) {
-      return this.parseSyntaxError_('Unsupported annotated expression');
+      this.reportError_(this.annotations_[0].location,
+                        'Unsupported annotated expression');
```
Comment (johnjbarton) [inline src/syntax/Parser.js]: Can we append the unsupported expression to the message here?

**Comment by johnjbarton** [comment]
LGTM, nice.

File: src/syntax/Parser.js (hunk 2 above)
Comment (arv) [inline src/syntax/Parser.js]: The location will tell you what it is.

File: src/syntax/Parser.js (hunk 1 above)
Comment (arv) [inline src/syntax/Parser.js]: Done.

**Pull request merged**